src/Nethermind/Nethermind.TxPool/TxPool.cs

```diff
@@ -825,6 +823,11 @@ public bool RemoveTransaction(Hash256? hash)
             return hasBeenRemoved;
         }
 
+        public bool ContainsTx(Hash256 hash)
+            => _blobTransactions.ContainsKey(hash) ||
```

Comment:
Eth68 is a requirement for blobs support so we shouldn't land in this part of code with blob tx

Reply:
Good question. We shouldn't request them.
Potential reason - we are not validating received transactions - everyone can send blob tx to us and we will accept it, even if not requested.
Adding such validation is in my backlog, about 1,5 year already

Reply:
It should work, we shouldn't request same tx multiple times.
But if we receive unrequested blob txs, we should probably disconnect peer and we are not doing it.

Reply:
Or if you mean why we need to check both cache (IsKnown) and TxPool, below eth68 we actually don't need, as all txs are in memory.
After eth68 (so with blobs) there is a scenario, when node was restarted and txs were loaded from disc. In this case we don't have them in cache

Reply:
Maybe we should load them on start

Reply:
There seem to be multiple hashes for blob tx?

Does something different if in mempool for etc?